bundles/org.openhab.binding.smarther/src/main/java/org/openhab/binding/smarther/internal/api/SmartherApi.java

```diff
@@ -0,0 +1,481 @@
+     * @throws {@link SmartherGatewayException}
+     *             in case of communication issues with the API gateway
+     */
+    public ModuleStatus getModuleStatus(String plantId, String moduleId) throws SmartherGatewayException {
+        final ContentResponse response = requestModule(GET, plantId, moduleId, null);
+        return ModelUtil.gsonInstance().fromJson(response.getContentAsString(), ModuleStatus.class);
+    }
+
+    /**
+     * Sends new settings to be applied to a given chronothermostat module.
+     *
+     * @param settings
+     *            the module settings to be applied
+     *
+     * @return {@code true} if the settings have been successfully applied, {@code false} otherwise
+     *
+     * @throws {@link SmartherGatewayException}
+     *             in case of communication issues with the API gateway
+     */
+    public boolean setModuleStatus(ModuleSettings settings) throws SmartherGatewayException {
+        // Prepare request payload
+        Map<String, Object> rootMap = new IdentityHashMap<String, Object>();
+        rootMap.put(ATTR_FUNCTION, settings.getFunction().getValue());
+        rootMap.put(ATTR_MODE, settings.getMode().getValue());
+        switch (settings.getMode()) {
+            case AUTOMATIC:
+                // {"function":"heating","mode":"automatic","programs":[{"number":0}]}
+                Map<String, Integer> programMap = new IdentityHashMap<String, Integer>();
+                programMap.put(ATTR_NUMBER, Integer.valueOf(settings.getProgram()));
+                List<Map<String, Integer>> programsList = new ArrayList<>();
+                programsList.add(programMap);
+                rootMap.put(ATTR_PROGRAMS, programsList);
+                break;
+            case MANUAL:
+                // {"function":"heating","mode":"manual","setPoint":{"value":0.0,"unit":"C"},"activationTime":"X"}
+                QuantityType<Temperature> newTemperature = settings.getSetPointTemperature(SIUnits.CELSIUS);
+                if (newTemperature == null) {
+                    throw new SmartherGatewayException("Invalid temperature unit transformation");
+                }
+                Map<String, Object> setPointMap = new IdentityHashMap<String, Object>();
+                setPointMap.put(ATTR_VALUE, newTemperature.doubleValue());
+                setPointMap.put(ATTR_UNIT, MeasureUnit.CELSIUS.getValue());
+                rootMap.put(ATTR_SETPOINT, setPointMap);
+                rootMap.put(ATTR_ACTIVATION_TIME, settings.getActivationTime());
+                break;
+            case BOOST:
+                // {"function":"heating","mode":"boost","activationTime":"X"}
+                rootMap.put(ATTR_ACTIVATION_TIME, settings.getActivationTime());
+                break;
+            case OFF:
+                // {"function":"heating","mode":"off"}
+                break;
+            case PROTECTION:
+                // {"function":"heating","mode":"protection"}
+                break;
```

Comment:
missing implementations?

Reply:
No :) simply OFF and Protection mode don't require additional attributes to be set, I've just kept them listed there to remind me and other potential developers that those cases exist and what is their current mapping required by API (in the comment).
It shouldn't be a problem to keep them, correct?